ui/pages/confirmations/components/confirm/info/dapp-swap-comparison-utils.ts

```diff
@@ -0,0 +1,167 @@
+import { Hex } from '@metamask/utils';
+import { Interface, TransactionDescription } from '@ethersproject/abi';
+import { QuoteResponse } from '@metamask/bridge-controller';
+import { addHexPrefix } from 'ethereumjs-util';
+import { getNativeTokenAddress } from '@metamask/assets-controllers';
+
+import { Numeric } from '../../../../../../shared/modules/Numeric';
+
+export const ABI = [
+  {
+    constant: true,
+    inputs: [
+      {
+        name: 'commands',
+        type: 'bytes',
+      },
+      {
+        name: 'inputs',
+        type: 'bytes[]',
+      },
+      {
+        name: 'deadline',
+        type: 'uint256',
+      },
+    ],
+    name: 'execute',
+    type: 'function',
+  },
+  {
+    constant: true,
+    inputs: [
+      {
+        name: 'commands',
+        type: 'bytes',
+      },
+      {
+        name: 'inputs',
+        type: 'bytes[]',
+      },
+    ],
+    name: 'execute',
+    type: 'function',
+  },
+];
+
+const getWordsFromInput = (input: string) => {
+  return input?.slice(2).match(/.{1,64}/gu) ?? [];
+};
+
+const wordToAddress = (word?: string) => {
+  if (!word) {
+    return '';
+  }
+  return addHexPrefix(word?.slice(24));
+};
+
+const wordToAmount = (word?: string) => {
+  if (!word) {
+    return '';
+  }
+  const amount = word?.replace(/^0+/u, '') ?? '';
+  return addHexPrefix(amount);
+};
+
+const parseTransactionData = (data?: string) => {
+  const contractInterface = new Interface(ABI);
+
+  let parsedTransactionData: TransactionDescription;
+
+  try {
+    parsedTransactionData = contractInterface.parseTransaction({
+      data: data as Hex,
+    });
+  } catch (error) {
+    return { inputs: [], commandBytes: [] };
+  }
+
+  const { commands } = parsedTransactionData.args;
+  const { inputs } = parsedTransactionData.args;
+  const commandBytes = commands.slice(2).match(/.{1,2}/gu) ?? [];
+
+  return { inputs, commandBytes };
+};
+
+export const getDataFromSwap = (
```

Comment:
Minor, if these are all standard functions, could we use `function` with this at the top for readability?